Add wxGrid::CopySelection()

Blake-Madden opened the pull request:
The was the same code from CTRL+C. I just refactored it into a separate function that the client can call now.

Blake-Madden committed:
Add wxGrid::CopySelection()
The was the same code from CTRL+C. I just refactored it into a separate function that the client can call now.

vadz commented on src/generic/grid.cpp:
```diff
@@ -10852,6 +10787,77 @@ void wxGrid::ClearSelection()
+        {
+            // As we use simple text format, we can't copy more
+            // than one block to clipboard.
+            wxLogWarning
+            (
+                _("Copying more than one selected block "
+                    "to clipboard is not supported.")
+            );
+            return;
```
I think it's not very helpful for the new function to give this warning, so I'd consider changing the code further to make the new function return `bool` indicating whether anything has been copied or not and giving this message in the original event handler (alternative: make the function return the error message, as string, with empty string meaning "no error", but it looks like an overkill here).

If not, the documentation should at least mention that the function logs a warning if there is more than one selected block.

vadz reviewed (commented):
Thanks, extracting this function looks like a good idea, but I'm not sure about the warning.

Blake-Madden committed:
Make CopySelection() return false upon error, remove log message

vadz commented on src/generic/grid.cpp:
```diff
@@ -6095,72 +6095,7 @@ void wxGrid::OnKeyDown( wxKeyEvent& event )
-                    }
-
-                    wxTheClipboard->SetData(new wxTextDataObject(buf));
+                    CopySelection();
```
Sorry, I think there was a misunderstanding, I meant to do this:
```suggestion
                    if ( !CopySelection() )
                    {
                        wxLogWarning(...);
                    }
```

vadz reviewed (commented):
Sorry, I didn't meant to remove the warning, just move it out from this function.

Blake-Madden committed:
Log copy error from CTRL+C, remove misleading comment

Blake-Madden commented on src/generic/grid.cpp:
```diff
@@ -10852,6 +10787,74 @@ void wxGrid::ClearSelection()
+    if (!lockClipboard)
+    {
+        // Error message should have been already given and we
+        // don't have much to add.
```
I removed this and make the CTRL+C log message more encompassing (but still informative).

Blake-Madden committed:
Trailing trailing newline from copied grid

This fixes the trailing newline when copy the grid. This newline is a bit annoying when you just copy one cell.

vadz commented:
Thanks, looks good to me, will merge in a moment.

vadz commented:
Sorry, after further thought I've realized that I am not sure with the latest commit added here. While I agree that it's useless when copying just a single cell, I think the trailing newline should be preserved when copying multiple cells.

So for now I've (squash) merged the rest, but please resubmit the latest commit as another PR preserving the multiple cells case unless you have a convincing reason to remove it in that case too.

TIA!
